spec/ruby/core/io/timeout_spec.rb

```diff
@@ -0,0 +1,104 @@
+# -*- encoding: utf-8 -*-
+require_relative '../../spec_helper'
+
+describe "IO#timeout" do
+  before :each do
+    @fname = tmp("io_timeout.txt")
+    @file = File.open(@fname, "a+")
+
+    @rpipe, @wpipe = IO.pipe
+    # There is no strict long term standard for pipe limits (2**16 bytes currently). This is an attempt to set a safe
+    # enough size to test a full pipe.
+    @more_than_pipe_limit = 1 << 18
+  end
+
+  after :each do
+    @rpipe.close
+    @wpipe.close
+
+    @file.close
+    rm_r @fname
+  end
+
+  ruby_version_is "3.2" do
```

Comment:
This should be the outermost block, because IO#timeout only exists on 3.2+

Reply:
Moved.